lib/rules/disallow-keywords-in-comments.js

```diff
@@ -0,0 +1,131 @@
+/**
+ * Disallows keywords in your comments, such as TODO or FIXME
+ *
+ * Type: `Boolean` or `String` or `Array`
+ *
+ * Values:
+ * - `true`
+ * - `'\b(word1|word2)\b'`
+ * - `['word1', 'word2']`
+ *
+ * #### Examples
+ *
+ * ```js
+ * "disallowKeywordsInComments": true
+ * "disallowKeywordsInComments": '\b(word1|word2)\b'
+ * "disallowKeywordsInComments": ['word1', 'word2']
+ * ```
+ *
+ * #### Invalid:
+ * ```
+ * // ToDo
+ * //TODO
+ * /** fixme *\/
+ * /**
+ *  * FIXME
+ *  *\/
+ * ```
+ */
+
+var assert = require('assert');
+
+// default keywords
+var keywords = ['todo', 'fixme'];
+
+var keywordRegEx;
+var buildKeywordRegEx = function(suppliedPattern) {
+    var pattern;
+
+    if (!suppliedPattern) {
+        pattern = '\\b(';
+        pattern += keywords.join('|');
+        pattern += ')\\b';
+    }
+    else {
+      pattern = suppliedPattern;
+    }
+
+    return new RegExp(pattern, 'gi');
+};
+
+var message;
+var buildMessage = function() {
+    var keywordsList = keywords.join(', ');
+    return 'Comments cannot contain the following keywords: ' + keywordsList;
+};
+
+var getCommentErrors = function(comment) {
+    var splitComment = comment.value.split('\n');
+    var errorArray = [];
+
+    splitComment.forEach(function(commentNode, index) {
+        var lineIndex = index;
+        var matches = commentNode.match(keywordRegEx);
+        var lastIndex = 0;
+
+        if (matches) {
+            matches.map(function(match) {
+                var foundAtIndex;
+
+                lastIndex = lastIndex === 0 ? lastIndex : lastIndex + 1;
+                foundAtIndex = commentNode.indexOf(match, lastIndex);
+                lastIndex = foundAtIndex;
+
+                // line + lineIndex because comment block was split at new lines
+                //   will place carat at correct place within multiline comment
+                // foundAtIndex += 2 because comment opening is stripped
+                //   +2 finds accurate carat position on opening line comment
+                errorArray.push({
+                  line: comment.loc.start.line + lineIndex,
+                  column: lineIndex <= 0 ? foundAtIndex += 2 : foundAtIndex
+                });
+            });
+        }
+    });
+
+    return errorArray;
+};
+
+
+module.exports = function() {};
+
+module.exports.prototype = {
+    configure: function(disallowKeywordsInComments) {
+        // use default keywords
+        if (disallowKeywordsInComments === true) {
+            message = buildMessage();
```

Comment:
We typically store data on the rule instance to avoid shared state across instances: `this._message = buildMessage()`.